pkg/prometheus/server/statefulset.go

```diff
@@ -246,7 +246,7 @@ func makeStatefulSetSpec(
 	if cpf.TSDB != nil && cpf.TSDB.OutOfOrderTimeWindow != nil &&
 		compactionDisabled(p) &&
 		cg.WithMinimumVersion("2.55.0").IsCompatible() {
-		promArgs = append(promArgs, monitoringv1.Argument{Name: "storage.tsdb.allow-overlapping-compaction"})
+		promArgs = append(promArgs, monitoringv1.Argument{Name: "no-storage.tsdb.allow-overlapping-compaction"})
```

Comment:
I could be wrong but I don't see this flag in https://github.com/prometheus/prometheus/blob/3602785a89162ccc99a940fb9d862219a2d02241/cmd/prometheus/main.go#L455

Reply:
Thank you for clarifying :)